File: dataset/human/cpp/file_2463.cpp
#include <iostream>

int main()
{
    int N;
    long long A[100000];
    std::cin >> N;

    bool zero = false;
    int minus = 0;
    for (int i = 0; i < N; ++i)
    {
        std::cin >> A[i];
        if (A[i] == 0)
        {
            zero = true;
        }
        else if (A[i] < 0)
        {
            minus++;
        }
    }

    long long ans = 0;
    if (zero || (minus % 2) == 0)
    {
        for (int i = 0; i < N; ++i)
        {
            if (A[i] < 0)
            {
                ans += A[i] * -1;
            }
            else
            {
                ans += A[i];
            }
        }
    }
    else
    {
        long long min = 1e9 + 1;
        for (int i = 0; i < N; ++i)
        {
            int a = A[i];
            if (a < 0)
            {
                a *= -1;
            }

            ans += a;
            if (a < min)
            {
                min = a;
            }
        }

        ans -= min * 2;
    }

    std::cout << ans << std::endl;

    return 0;
}

File: dataset/human/python/file_2463.py
def resolve():
    d,g = map(int,input().split())
    pc = [list(map(int, input().split())) for _ in range(d)]
    ans = 10**9
    for i in range(2**d):
        count = 0
        total = 0
        nokori = [1 for _ in range(d)]
        for j in range(d):
            if (i>>j & 1):
                total += pc[j][0] * (j+1)* 100 + pc[j][1]
                count += pc[j][0]
                nokori[j] = 0
        if total >= g:
            ans = min(ans, count)
        else:
            for k in reversed(range(len(nokori))):
                if nokori[k] == 0:
                    continue
                else:
                    if total + (pc[k][0]-1) * (k+1) *100 >= g:
                        count += (((g-total) + (k+1)*100)-1)//((k+1)*100)
                        ans = min(ans, count)
                        break
                    else:
                        total += (pc[k][0]-1) * (k+1) *100
                        count += (pc[k][1]-1)
    print(ans)
resolve()
